### PR TITLE
Refactor SDF handling logic for non-existing string paths

### PR DESCRIPTION
**Pull request by flferretti**
The non-existing paths will be handled by `pathlib.Path.read_text()`, eventually raising a `FileNotFoundError`.

### CODE REVIEW

**Review by traversaro — APPROVED** [review]

**Review by CarlottaSartore — APPROVED** [review]

**Comment by flferretti** [comment]
> I'm not 100% sure that the new logic covers entirely the old logic. Also, the previous refactoring might have introduced some unchecked cases.
> 
> It's probably worth to have a unit test for such a core component of the library like loading the supported types of resources.

I added additional tests and refactored forcing `is_urdf`

File: src/rod/sdf/sdf.py [hunk 1]
```diff
@@ -60,34 +60,30 @@ def load(sdf: pathlib.Path | str, is_urdf: bool | None = None) -> Sdf:
...
-                case _:
-                    raise TypeError(f"Unsupported type for 'sdf': {type(sdf)}")
+        # Check if the input is a string path.
+        is_str_path = getattr(sdf, "__len__", lambda: MAX_PATH + 1)() <= MAX_PATH
+
+        match sdf:
+            # Case 1: Handle SDF/URDF string.
+            case str() if not is_str_path:
+                sdf_string = sdf
+                is_urdf = is_urdf if is_urdf is not None else "<robot" in sdf_string
```
Comment (diegoferigo) [inline src/rod/sdf/sdf.py:76]: There could be SDF/URDF content shorter than MAX_PATH=260 characters right?

**Review by diegoferigo — COMMENTED** [review]
I think the three original cases were correct, and now something is missing:

1. If the resource is a path, expect that is exists and read its content.
2. If the resource is a string:
  - a) if its length is shorter than the max path length, check if it's a file. If it is, read its content, otherwise treat it as a string containing the URDF/SDF.
  - b) if its length is longer than the max path length, treat is as a string containing the URDF/SDF.

How is 2a handled now? The logic seems changed.

**Comment by flferretti** [comment]
> I think the three original cases were correct, and now something is missing:
> 
>     1. If the resource is a path, expect that is exists and read its content.
> 
>     2. If the resource is a string:
> 
> 
>     * a) if its length is shorter than the max path length, check if it's a file. If it is, read its content, otherwise treat it as a string containing the URDF/SDF.
> 
>     * b) if its length is longer than the max path length, treat is as a string containing the URDF/SDF.
> 
> 
> How is 2a handled now? The logic seems changed.

Solved by rebase. I had to make the assumption that the passed string path has an extension, I couldn't find any other way to distinguish between a URDF shorter than MAX_PATH and a string path.

File: src/rod/sdf/sdf.py (hunk 1 above)
Comment (flferretti) [reply src/rod/sdf/sdf.py]: Solved by rebase

File: src/rod/sdf/sdf.py [hunk 2]
```diff
@@ -60,34 +59,28 @@ def load(sdf: pathlib.Path | str, is_urdf: bool | None = None) -> Sdf:
...
+        path = pathlib.Path(sdf)
 
-                # Case 3: It's an SDF/URDF string
-                case str():
+        match sdf:
+            # Case 1: Handle strings.
+            case str():
+                if path.suffix:
+                    # Assuming that if the string has a suffix, it is a path.
+                    sdf_string = pathlib.Path(sdf).read_text(encoding="utf-8")
```
[the remark this answers is not in the document]
Comment (flferretti) [reply src/rod/sdf/sdf.py:69]: Yes, it looks pretty clear
```python
In [2]: pathlib.Path("non/existing/path.sdf").read_text()
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[4], line 1
----> 1 pathlib.Path("non/existing/path.sdf").read_text()

File ~/miniforge3/envs/jaxsim/lib/python3.13/pathlib/_local.py:546, in Path.read_text(self, encoding, errors, newline)
    543 # Call io.text_encoding() here to ensure any warning is raised at an
    544 # appropriate stack level.
    545 encoding = io.text_encoding(encoding)
--> 546 return PathBase.read_text(self, encoding, errors, newline)

File ~/miniforge3/envs/jaxsim/lib/python3.13/pathlib/_abc.py:632, in PathBase.read_text(self, encoding, errors, newline)
    628 def read_text(self, encoding=None, errors=None, newline=None):
    629     """
    630     Open the file in text mode, read it, and close the file.
    631     """
--> 632     with self.open(mode='r', encoding=encoding, errors=errors, newline=newline) as f:
    633         return f.read()

File ~/miniforge3/envs/jaxsim/lib/python3.13/pathlib/_local.py:537, in Path.open(self, mode, buffering, encoding, errors, newline)
    535 if "b" not in mode:
    536     encoding = io.text_encoding(encoding)
--> 537 return io.open(self, mode, buffering, encoding, errors, newline)

FileNotFoundError: [Errno 2] No such file or directory: 'non/existing/path.sdf'
```

File: src/rod/sdf/sdf.py [hunk 3]
```diff
@@ -60,34 +59,28 @@ def load(sdf: pathlib.Path | str, is_urdf: bool | None = None) -> Sdf:
...
-                    is_urdf = "<robot" in sdf_string
 
-                # Case 4: Raise an error for unsupported types
-                case _:
-                    raise TypeError(f"Unsupported type for 'sdf': {type(sdf)}")
+                is_urdf = is_urdf if is_urdf is not None else "<robot" in sdf_string
+
+            # Case 2: Handle pathlib.Path.
+            case pathlib.Path():
+                sdf_string = path.read_text(encoding="utf-8")
```
[the remark this answers is not in the document]
Comment (flferretti) [reply src/rod/sdf/sdf.py:78]: Solved in https://github.com/ami-iit/rod/pull/50#discussion_r1837933178

File: src/rod/sdf/sdf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (diegoferigo) [reply src/rod/sdf/sdf.py:69]: Great!

**Pull request merged**